apps/voting-disputable/contracts/DisputableVoting.sol

```diff
@@ -0,0 +1,745 @@
+    */
+    function _voteOnBehalfOf(uint256 _voteId, bool _supports, address _voter, address _representative) internal {
+        Vote storage vote_ = votes[_voteId];
+        _castVote(vote_, _voteId, _supports, _voter);
+        vote_.casters[_voter] = _representative;
+
+        emit ProxyVoteSuccess(_voteId, _voter, _representative, _supports);
+    }
+
+    /**
+    * @dev Internal function to execute a vote. It assumes the queried vote exists.
+    */
+    function _executeVote(uint256 _voteId) internal {
+        require(_canExecute(votes[_voteId]), ERROR_CAN_NOT_EXECUTE);
+        _unsafeExecuteVote(_voteId);
+    }
+
+    /**
+    * @dev Internal unsafe version of _executeVote that assumes you have already checked if the vote can be executed and exists
+    */
+    function _unsafeExecuteVote(uint256 _voteId) internal {
+        Vote storage vote_ = votes[_voteId];
+
+        vote_.executed = true;
+        vote_.disputableStatus = DisputableStatus.Closed;
+        _closeAgreementAction(vote_.actionId);
+
+        bytes memory input = new bytes(0); // TODO: Consider input for voting scripts
+        address[] memory blacklist = new address[](1);
+        blacklist[0] = address(_getAgreement());
+        runScript(vote_.executionScript, input, blacklist);
+        emit ExecuteVote(_voteId);
+    }
+
+    /**
+    * @dev Challenge a vote
+    * @param _voteId Identification number of the vote to be challenged
+    */
+    function _onDisputableActionChallenged(uint256 _voteId, uint256 /* _challengeId */, address /* _challenger */) internal {
+        Vote storage vote_ = votes[_voteId];
+        require(_canPause(vote_), ERROR_CANNOT_PAUSE_VOTE);
+
+        vote_.disputableStatus = DisputableStatus.Paused;
+        vote_.pausedAt = getTimestamp64();
+        emit PauseVote(_voteId);
+    }
+
+    /**
+    * @dev Allow a vote
+    * @param _voteId Identification number of the vote to be allowed
+    */
+    function _onDisputableActionAllowed(uint256 _voteId) internal {
+        Vote storage vote_ = votes[_voteId];
+        require(_isPaused(vote_), ERROR_VOTE_NOT_PAUSED);
+
+        vote_.disputableStatus = DisputableStatus.Active;
+        vote_.pauseDuration = getTimestamp64().sub(vote_.pausedAt);
+        emit ResumeVote(_voteId);
+    }
+
+    /**
+    * @dev Reject a vote
+    * @param _voteId Identification number of the vote to be rejected
+    */
+    function _onDisputableActionRejected(uint256 _voteId) internal {
+        Vote storage vote_ = votes[_voteId];
+        require(_isPaused(vote_), ERROR_VOTE_NOT_PAUSED);
+
+        vote_.disputableStatus = DisputableStatus.Cancelled;
+        vote_.pauseDuration = getTimestamp64().sub(vote_.pausedAt);
+        emit CancelVote(_voteId);
+    }
+
+    /**
+    * @dev Void an entry
+    * @param _voteId Identification number of the entry to be voided
+    */
+    function _onDisputableActionVoided(uint256 _voteId) internal {
+        _onDisputableActionAllowed(_voteId);
```

Comment:
Haha this is a little bit misleading, especially with the natspec.

Perhaps we can write some comments here to explain why we're choosing `allowed`.